enhancements/okd-on-fedora-coreos.md

```diff
@@ -32,128 +32,58 @@ superseded-by:
-
-This is implemented by this PR: https://github.com/openshift/machine-config-operator/pull/1703
-
-The last commit in that PR already shows that the spec used to store on-disk state internally (i.e. the rendered machine config) could already be switched over to v3 without any breaking changes in external APIs. This however is not required to enable using the same MCO images for OKD and OCP.
-
-### 3: Sample Operators Changes
-
-#### Mark RHEL based operator images as optional
-
-The samples operator degrades on OKD due to the fact that not all images are pullable without a valid pull secret. The behaviour should change so it does not degrade on OKD clusters for users that have not specified a valid pull secret.
-
-Alternatively, all RHEL based images this concers could be rebuilt on a CentOS or Fedora base, and a mechanism to pull them in case of no valid pull secret could be implemented in `cluster-samples-operator`. 
-
-### 4: Installer Changes
-
-These changes are optional for the immediate goal of releasing OKD 4 GA. They should however be strongly considered in order to facilitate dropping the OKD fork of installer rather sooner than later.
-
-#### Add fcos.json
-
-Support for an `fcos.json` to exist alongside `rhcos.json` for OS image references.
-
-This could be enabled either via a build-time flag to produce seperate OKD installer artifacts, or an `fcos` or `okd` run-time flag to pass to the installer for it to provision Fedora CoreOS instead of RHEL CoreOS hosts (in that case it would default to contents of `rhcos.json`). 
-
-#### Add option to disable pull secret to Installer
-
-OKD installs do not require a pull secret, so make it optional.
-
-This could be enabled either via a build-time flag to produce seperate OKD installer artifacts, or a run-time flag to signal to the installer to not require a pull secret.
-
-#### Switching Installer to Ignition spec v3
-
-The MachineConfig API vendored in the Installer has to be updated to the current version that uses a RawExtension to store the ignition config.
-
-To facilitate the migration, all functions requiring Ignition types will be abstracted into one package. This is implemented by this PR: https://github.com/openshift/installer/pull/3248
-
-With the MCO's ability to process both spec v2 and v3 configs, switching the installer over to spec v3 is technically possible.
-
-It is imperative that the timing of this change is aligned with RHEL CoreOS images switching to an Ignition v2.x binary that supports spec v3.
-
-Once this migration is complete, it will be possible for OKD to use an installer built off of the same sources.
+This is a proposal to:
+- Create all OKD artifacts in OpenShift's Prow instance from the canonical OpenShift master branch and the latest release branch for testing,
```

Comment:
We create OKD from all active branching (even 4.3 nightlies are being produced). However we'll promote to stable only the latest supported branch (`release-4.6` at the time). Not sure if that needs to be clarified

Reply:
May be it should be `master branch and at least the latest release branch`.

In the future, we could possible remove the building of versions before that to save CI costs - there's nobody consuming these artifacts after all.

Reply:
>master branch and at least the latest release branch

This sounds good.

> In the future, we could possible remove the building of versions before that to save CI costs - there's nobody consuming these artifacts after all.

IIUC we're keeping last 5 artifacts, regardless of the age, so it shouldn't use too much resources. Promotion jobs are still running once a day for "old stable" streams, so it shouldn't affect CI too much